File: 프로그래머스/lv2/131127. 할인 행사/할인 행사.java
import java.util.*;
class Solution {
    public int solution(String[] want, int[] number, String[] discount) {
        int answer = 0;
        HashMap<String, Integer> map = new HashMap<>();
        
        for(int i=0;i<want.length;i++)
            map.put(want[i], number[i]);

        for(int i=0;i<=discount.length-10;i++){
            HashMap<String, Integer> tmp = new HashMap<>();
            for(int j=i;j<i+10;j++){
                tmp.put(discount[j],tmp.getOrDefault(discount[j],0)+1);
            }
          
            boolean join = true;
            for(String s: map.keySet()){
                if(tmp.getOrDefault(s,0)<map.get(s)){
                    join = false;
                    break;
                }
            }
            if(join)    answer++;
        }
        
        return answer;
    }
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
def solution(want, number, discount):
    answer = 0
    pd={}
    for i in range(len(want)):
        pd[want[i]]=i

    for i in range(len(discount)-9):
        cnts=[0]*len(want)
        for j in range(i,i+10):
            if discount[j] in pd:
                cnts[pd[discount[j]]]+=1
        possible=True
        for j in range(len(want)):
            if number[j]>cnts[j]:
                possible=False
                break
        if possible:    answer+=1
    return answer